src/ast/ClassDecl.ts

```diff
@@ -74,6 +75,23 @@ export class ClassDecl extends Content {
         this.functions.forEach((funcDecl: FuncDecl) => funcDecl.typeCheck());
     }
 
+    public fulfillContract(): void {
```

Comment:
if class A extends abstract class B, then A needs to implement B's methods right? this case seems to be missing

Reply:
Only if those methods are declared `abstract` - I'll check and see if we support abstract function declarations in our DSL.

Reply:
Nope!

We only support abstract classes, but (strangely) LOL - `FuncDecl` doesn't allow for `abstract`.

Reply:
ok everythings gucci

Reply:
nice to have? (abstract functions)

Reply:
@kiyomih sounds like it